scipy/linalg/basic.py

```diff
@@ -343,8 +343,12 @@ def solve_triangular(a, b, trans=0, lower=False, unit_diagonal=False,
         print('solve:overwrite_b=', overwrite_b)
     trans = {'N': 0, 'T': 1, 'C': 2}.get(trans, trans)
     trtrs, = get_lapack_funcs(('trtrs',), (a1, b1))
-    x, info = trtrs(a1, b1, overwrite_b=overwrite_b, lower=lower,
-                    trans=trans, unitdiag=unit_diagonal)
+    if a1.flags.f_contiguous:
+        x, info = trtrs(a1, b1, overwrite_b=overwrite_b, lower=lower,
+                        trans=trans, unitdiag=unit_diagonal)
+    else:
```

Comment:
Please also consider adding a comment that `trtrs` expects Fortran's column-major ordering, and rather than actually copying the array into Fortran order, you instruct `trtrs` to solve the transposed system.